services/hack/getHacks.js

```diff
@@ -18,38 +20,44 @@ export const list = async (event, context) => {
           from: "users",
           localField: "team",
           foreignField: "_id",
-          as: "team"
-        }
+          as: "team",
+        },
       },
       {
         $project: {
           team: { name: 1, _id: 1 },
           likes: { $size: "$likes" },
           description: 1,
           _id: 1,
+          hasUserLiked: {
+            $in: [
+              userId !== "null" ? mongoose.Types.ObjectId(userId) : null,
```

Comment:
I'm not liking that were only checking for a string "null" and not an actual `null`, or empty string or other falsey values.

Reply:
Added new condition to guard against any other falsy values for userId

The reason we are guarding the string "null" is because the front end sends the GET request with **/hacklists?userId=null** the "null" is the initialstate of the redux store when the user has not been authenticated, javascript auto-convert null to string. So at the backend we have to take into account of that as well as other falsy values.

```javascript
hasUserLiked: {
            $in: [
              userId !== "null" || !userId
                ? mongoose.Types.ObjectId(userId)
                : null,
              "$likes",
            ],
          },
```